packages/typespec-ts/src/modular/buildProjectFiles.ts

```diff
@@ -524,14 +504,10 @@ export function emitTsConfig(
   );
 
   const { packageDetails, azureSdkForJs } = codeModel.options || {};
-  let { generateTest, generateSample } = codeModel.options || {};
+  const { generateTest, generateSample } = codeModel.options || {};
   const isBranded = codeModel.options?.branded ?? true;
   // Take the undefined as true by default
-  generateTest = generateTest === true || generateTest === undefined;
-  generateSample =
-    (generateSample === true || generateSample === undefined) &&
-    hasSamplesGenerated;
-  const clientPackageName = packageDetails?.name ?? "@msinternal/unamedpackage";
+  const clientPackageName = packageDetails!.name;
```

Comment:
is it possible to be undefined for packageDetails?

Reply:
no, we will return default value for packageDetails as here https://github.com/Azure/autorest.typescript/pull/2296/files#diff-302d163c5f1ccfffe405495b95dba123e3c5f70e537a0186fa265eb168c1a225R250-R256